packages/gatsby/src/joi-schemas/joi.js

```diff
@@ -1,10 +1,18 @@
 const Joi = require(`joi`)
 
+const stripTrailingSlash = chain => chain.replace(/(\w)\//, `$1`)
+
 export const gatsbyConfigSchema = Joi.object().keys({
   __experimentalThemes: Joi.array(),
   polyfill: Joi.boolean(),
   siteMetadata: Joi.object(),
-  pathPrefix: Joi.string(),
+  assetPrefix: stripTrailingSlash(Joi.string().uri()),
```

Comment:
This _could possibly_ be a relative URL too--so will investigate as to whether that makes sense.

Reply:
Our use case would definitely benefit from using a relative URL.

We're using a Cloudflare worker to serve content from a Gatsby site for certain routes. Our issue right now is that the assets are dumped into the root of `public`. We've worked around this in Gatsby v1 using `--prefix-paths` and having our worker serve Gatsby files for the `prefixPath`.

With Gatsby v2 this is also prefixing our routes, which causes our homepage to redirect to our `prefixPath`. We could certainly work around having provide a full URI, but it seems relative would be easier for us.

After adding `allowRelative: true` here, it seems to work well for our use case.

Reply:
@DanielBrierton did you actually check this feature out--or do you mean that the relative URL validation will work for your use case?

May be helpful if we could get together this week or next and see if this feature works for your use case, and actually get something deployed!

Reply:
Yeah, I checked out your PR and made the change to actually test it. I haven't tested it in our production/staging environment, but I'm confident it will work based off how it worked in my local dev env.

Happy to help testing it. We were originally going to make a plugin based on https://github.com/gatsbyjs/gatsby/issues/8833#issuecomment-458147983 (which works too), but an actual supported feature in Gatsby would obviously be much more preferable.

Reply:
relativeUrls for assetPrefix work just fine! Albeit with the gotcha that it can't be used in conjunction with a relative pathPrefix, as well!